database/postgres/postgres.go

```diff
@@ -332,20 +332,19 @@ func (p *Postgres) Drop() error {
 	return nil
 }
 
-func (p *Postgres) ensureVersionTable() error {
-	// check if migration table exists
-	var count int
-	query := `SELECT COUNT(1) FROM information_schema.tables WHERE table_name = $1 AND table_schema = (SELECT current_schema()) LIMIT 1`
-	if err := p.conn.QueryRowContext(context.Background(), query, p.config.MigrationsTable).Scan(&count); err != nil {
-		return &database.Error{OrigErr: err, Query: []byte(query)}
-	}
-	if count == 1 {
-		return nil
+func (p *Postgres) ensureVersionTable() (err error) {
+	if err = p.Lock(); err != nil {
+		return err
 	}
 
-	// if not, create the empty migration table
-	query = `CREATE TABLE "` + p.config.MigrationsTable + `" (version bigint not null primary key, dirty boolean not null)`
-	if _, err := p.conn.ExecContext(context.Background(), query); err != nil {
+	defer func() {
+		if e := p.Unlock(); err == nil {
+			err = e
+		}
+	}()
+
+	query := `CREATE TABLE IF NOT EXISTS "` + p.config.MigrationsTable + `" (version bigint not null primary key, dirty boolean not null)`
```

Comment:
A note about the introduction of "IF NOT EXISTS" here. Without it, but with the advisory lock in place, we get these errors, which makes sense: https://gist.github.com/tsenart/3cb026aa5705d7a2db14ddc610a2a8d2

However, as per Postgres' documentation, it seems we wouldn't support versions older than https://www.postgresql.org/docs/9.1/sql-createtable.html, which is when "CREATE IF NOT EXISTS" was introduced.

If that's not desirable, we can revert to use a SELECT first to check if the table exists and then run the CREATE if not, but all wrapped in an advisory lock.

Reply:
Answered my own question. Reverted back to using a SELECT first.